# Percentage of Retweets and Tweets That are Just Links

In [26]:
import pandas as pd
import json
import re

trump_tweets = pd.read_json('since_inauguration_includes_retweets_01-20-17_05-27-20.json')

print(f"Total tweets since Inauguration (1/20/2017 - 5/27/2020): {len(trump_tweets)}")

Total tweets since Inauguration (1/20/2017 - 5/27/2020): 18468


In [27]:
tweets = trump_tweets["text"].tolist()

In [35]:
retweet_count = 0
for tweet in tweets:
    if(tweet.find('RT') == 0):
        retweet_count += 1
        
tweet_links_count = 0
for tweet in tweets:
    if(tweet.find('http') == 0):
        tweet_links_count += 1  

retweets_and_links = retweet_count + tweet_links_count        
        
print(f"Retweets since Inauguration (1/20/2017 - 5/27/2020): {retweet_count}")
print(f"Tweets that are not Retweets but are just links (1/20/2017 - 5/27/2020): {tweet_links}")
print(f"Total retweets and links: {retweets_and_links}")


Retweets since Inauguration (1/20/2017 - 5/27/2020): 6311
Tweets that are not Retweets but are just links (1/20/2017 - 5/27/2020): 507
Total retweets and links: 6818


In [36]:
percent_retweeted = int((retweets_and_links / len(trump_tweets)) * 100)
print(f"Percentage of retweets: {percent_retweeted}%")

Percentage of retweets: 36%


## Summary

Since Trump's Inauguration, we see that 36% of his tweets are retweets and tweets with only links